tango/executor.py

```diff
@@ -50,13 +52,35 @@ def execute_step_graph(self, step_graph: StepGraph) -> str:
 
         ordered_steps = sorted(step_graph.values(), key=lambda step: step.name)
 
+        # find uncacheable leaf steps
+        interior_steps: Set[Step] = set()
+        for step in ordered_steps:
+            for dependency in step.dependencies:
+                interior_steps.add(dependency)
+        uncacheable_leaf_steps = {
+            step for step in set(step_graph.values()) - interior_steps if not step.cache_results
+        }
+        # also pick any uncacheable direct dependencies of leaf steps
```

Comment:
Is that necessary? Won't those dependencies run anyway when we run the steps that depend on them?

Reply:
You are 100% right.

Reply:
21f0280